input/size.go

```diff
@@ -222,12 +234,30 @@ const (
 	// BaseAnchorCommitmentTxWeight 900 weight.
 	BaseAnchorCommitmentTxWeight = witnessScaleFactor * BaseAnchorCommitmentTxSize
 
+	// BaseTaprootCommitmentTxSize 125 + 43 * num-htlc-outputs bytes
+	//	- Version: 4 bytes
+	//	- WitnessHeader <---- part of the witness data
+	//	- CountTxIn: 1 byte
+	//	- TxIn: 41 bytes
+	//		FundingInput
+	//	- CountTxOut: 1 byte
+	//	- TxOut: 74 + 43 * num-htlc-outputs bytes
+	//		OutputPayingToThem,
+	//		OutputPayingToUs,
+	//		....HTLCOutputs...
+	//	- LockTime: 4 bytes
+	BaseTaprootCommitmentTxWeight = 4 + 1 + FundingInputSize + 3 +
+		2*TaprootCommitmentOutput + 2*TaprootCommitmentAnchorOutput + 4
+
 	// CommitWeight 724 weight.
 	CommitWeight = BaseCommitmentTxWeight + WitnessCommitmentTxWeight
 
 	// AnchorCommitWeight 1124 weight.
 	AnchorCommitWeight = BaseAnchorCommitmentTxWeight + WitnessCommitmentTxWeight
 
+	// TaprootCommitWeight 1124 weight.
+	TaprootCommitWeight = BaseTaprootCommitmentTxWeight + TaprootKeyPathWitnessSize
```

Comment:
```
  TaprootSignatureWitnessSize = 1 + 64
 TaprootKeyPathWitnessSize = 1 + TaprootSignatureWitnessSize
```

So it has the first byte for the number of elements, then one for the size of the stack element.

Reply:
I mean the `WitnessHeaderSize` which is the witness flag in the transaction

Reply:
I think adding it there would cause methods like `AddTaprootKeySpendInput` (weight estimator) to over account for the weight: the witness header only exists in a single place in the transaction, and isn't required for each added input.

Reply:
Tacked it onto TaprootCommitWeight